tests/components/MapboxMap.test.tsx

```diff
@@ -0,0 +1,112 @@
+import { render } from '@testing-library/react';
+import { mockAnswersState } from '../__utils__/mocks';
+import { CoordinateGetter, MapboxMap, Coordinate } from '../../src/components/MapboxMap';
+import { Source, State } from '@yext/search-headless-react';
+import { Map, Marker } from 'mapbox-gl';
+
+jest.mock('@yext/search-headless-react');
+jest.mock('mapbox-gl');
+
+interface Location {
+  customCoordinate: Coordinate
+}
+
+const mockedStateDefaultCoordinate: Partial<State> = {
+  vertical: {
+    verticalKey: 'vertical',
+    results: [{
+      rawData: {
+        yextDisplayCoordinate: {
+          latitude: 1,
+          longitude: 1
+        }
+      },
+      source: Source.KnowledgeManager
+    }]
+  }
+};
+
+const mockedStateCustomCoordinate: Partial<State> = {
+  vertical: {
+    verticalKey: 'vertical',
+    results: [{
+      rawData: {
+        customCoordinate: {
+          latitude: 2,
+          longitude: 2
+        }
+      },
+      source: Source.KnowledgeManager
+    }]
+  }
+};
+
+const mockedStateWrongCoordinateType: Partial<State> = {
+  vertical: {
+    verticalKey: 'vertical',
+    results: [{
+      rawData: {
+        yextDisplayCoordinate: [1, 1]
+      },
+      source: Source.KnowledgeManager
+    }]
+  }
+};
+
+describe('default "getCoordinate"', () => {
+  it('uses result\'s "yextDisplayCoordinate" for marker location', () => {
+    mockAnswersState(mockedStateDefaultCoordinate);
+    const setLngLat = jest.spyOn(Marker.prototype, 'setLngLat').mockReturnValue(Marker.prototype);
```

Comment:
could this spy/mock happen in a `beforeEach` to reduce repetition?

Reply:
I need access to the function being spied on in certain tests; I slightly prefer calling it in the individual tests then having a global `let setLngLat` and set it in beforeEach for the tests -- but I can change to it if you think that's fine

Reply:
oh I see that not all the tests are wrapped in this `describe`. I'm fine with leaving as is then